lib/src/dense_set.rs

```diff
@@ -0,0 +1,1282 @@
+            }
+        }
+        true
+    }
+
+    fn intersect(&mut self, other: &Self) {
+        let smallest_set_size = cmp::min(self.data.len(), other.data.len());
+        let mut res = Vec::<Reg>::new();
+
+        for &item in self.reg_vec.iter() {
+            if other.contains(item) {
+                res.push(item);
+            }
+        }
+        self.reg_vec = res;
+
+        for i in 0..smallest_set_size {
+            self.data[i] &= other.data[i];
+        }
+
+        for i in smallest_set_size..self.data.len() {
+            self.data[i] = 0;
+        }
+    }
+
+    fn union(&mut self, other: &Self) {
+        let smallest_set_size = cmp::min(self.data.len(), other.data.len());
+        let greatest_set_size = cmp::max(self.data.len(), other.data.len());
+
+        for &item in other.reg_vec.iter() {
+            if !self.contains(item) {
+                self.reg_vec.push(item);
+            }
+        }
+
+        for i in 0..smallest_set_size {
+            self.data[i] |= other.data[i];
+        }
+
+        if other.data.len() > self.data.len() {
+            for i in smallest_set_size..greatest_set_size {
+                self.data.push(other.data[i]);
+            }
+        }
+    }
+
+    fn remove(&mut self, other: &Self) {
+        let smallest_set_size = cmp::min(self.data.len(), other.data.len());
+
+        for &item in other.reg_vec.iter() {
+            if self.contains(item) {
+                self.vec_delete(item);
+            }
+        }
+
+        for i in 0..smallest_set_size {
+            self.data[i] &= !other.data[i];
+        }
+    }
+
+    fn intersects(&self, other: &Self) -> bool {
+        let smallest_set_size = cmp::min(self.data.len(), other.data.len());
+
+        for i in 0..smallest_set_size {
+            if self.data[i] & other.data[i] != 0 {
+                return true;
+            }
+        }
+        false
+    }
+
+    fn is_subset_of(&self, other: &Self) -> bool {
+        let smallest_set_size = cmp::min(self.data.len(), other.data.len());
+
+        for i in 0..smallest_set_size {
+            if (self.data[i] | other.data[i]) != other.data[i] {
+                return false;
+            }
+        }
+
+        if self.data.len() > other.data.len() {
+            for i in other.data.len()..self.data.len() {
+                if self.data[i] != 0 {
+                    return false;
+                }
+            }
+        }
+        true
+    }
+
+    fn equals(&self, other: &Self) -> bool {
+        self.data == other.data
+    }
+
+    fn vec_get_item(&self, index: usize, real: bool) -> Option<&Reg> {
+        for reg in self.reg_vec.iter() {
+            if reg.is_real() == real && reg.get_index() == index {
+                return Some(&reg);
+            }
+        }
+        None
+    }
+}
+
+impl fmt::Debug for BitSet {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{:?}", self.data)
+    }
+}
+
+impl Clone for BitSet {
```

Comment:
You can add `#[derive(Clone)]` at the top of the declaration of `BitSet`, to get the same effect.

Reply:
Thanks! It's done (055a40e).